src/main/kotlin/org/opensearch/commons/alerting/model/Comment.kt

```diff
@@ -37,21 +38,32 @@ data class Comment(
 
     constructor(
         entityId: String,
+        entityType: String,
         content: String,
         createdTime: Instant,
         user: User?
     ) : this (
         entityId = entityId,
+        entityType = entityType,
         content = content,
         createdTime = createdTime,
         lastUpdatedTime = null,
         user = user
     )
 
+    enum class EntityType(val value: String) {
```

Comment:
if we intend for comments to be used by other plugins we shouldn't declared EntityType as an Enum as XYZPlugin might not accept "Alert" as a valid enum and Alerting plugin might not accept XYZ as an entity type so the enum is just overloaded and would end up causing more validations in each plugin irrespective of its presence here